hyperspy/model.py

```diff
@@ -924,475 +956,698 @@ def red_chisq(self):
-                    de_b = tuple(((a if a is not None else -np.inf,
-                                   b if b is not None else np.inf) for a, b in de_b))
-                    self.p0 = differential_evolution(tominimize, de_b,
-                                                     args=args, **kwargs).x
+                            f"Bounds must be specified for fitter='{fitter}'"
+                        )
+
+                    de_b = tuple(
+                        (
+                            a if a is not None else -np.inf,
+                            b if b is not None else np.inf,
+                        )
+                        for a, b in self.free_parameters_boundaries
+                    )
+
+                    if np.any(~np.isfinite(de_b)):
+                        raise ValueError(
+                            "Finite upper and lower bounds must be "
+                            "specified for every free parameter"
+                        )
+
+                    self.fit_output = global_fs[fitter](
+                        f_min, de_b, args=args, **kwargs
+                    )
 
                 else:
-                    raise ValueError("""
-                    The %s optimizer is not available.
-
-                    Available optimizers:
-                    Unconstrained:
-                    --------------
-                    Least-squares: leastsq and odr
-                    General: Nelder-Mead, Powell, CG, BFGS, Newton-CG
-
-                    Constrained:
-                    ------------
-                    least_squares, mpfit, TNC and L-BFGS-B
-
-                    Global:
-                    -------
-                    Differential Evolution
-                    """ % fitter)
+                    self.fit_output = minimize(
+                        f_min,
+                        self.p0,
+                        jac=f_der,
+                        args=args,
+                        method=fitter,
+                        bounds=self.free_parameters_boundaries,
+                        **kwargs,
+                    )
+
+                self.p0 = self.fit_output.x
+                to_print.extend(["Fit result:", self.fit_output])
+
+                if fitter == "L-BFGS-B" and hasattr(self.fit_output, "hess_inv"):
+                    # For L-BFGS-B, we can place an upper bound on
+                    # parameter precision using the inverse Hessian,
+                    # since the optimization stops when (see docs):
+                    # (f^k - f^{k+1})/max{|f^k|,|f^{k+1}|,1} <= ftol
+
+                    # Default value of ftol for L-BFGS-B
+                    ftol = 2.220446049250313e-9
+                    if "options" in kwargs:
+                        ftol = kwargs["options"].get("ftol", ftol)
+
+                    cost = self.fit_output.fun
+                    hess = np.diag(self.fit_output.hess_inv.todense())
+                    self.p_std = np.sqrt(max(1.0, np.abs(cost)) * ftol * hess)
```

Comment:
The alternative is to just use `hess_inv.todense()` as an estimate of the covariance matrix and use the same procedure as `leastsq` - rather than do `np.sqrt(max(1.0, np.abs(cost)) * ftol * hess)`. But that's not particularly clean etc. So I'll remove it as you suggest. 

Taking the lmfit approach (or another) is probably best left for a separate PR.